File: 第十一次/11-1/Rectangle.cpp
#include"Rectangle.h"

Rectangle::Rectangle(double width, double height) {
	setWidth(width);
	setHeight(height);
}

void Rectangle::setWidth(double newWidht) {
	width = newWidht;
}
double Rectangle::getWidth() {
	return width;
}

void Rectangle::setHeight(double newHeight) {
	height = newHeight;
}
double Rectangle::getHeight() {
	return height;
}

double Rectangle::getArea() {
	return width * height;
}

void Rectangle::swapByReference(Rectangle &r2) {
	Rectangle temp = *this;
	*this = r2;
	r2 = temp;
}
void Rectangle::swapByPointer(Rectangle* r2) {
	Rectangle temp = *this;
	*this = *r2;
	*r2 = temp;

}

File: Python/第10章/10-2/Rectangle.py
class Rectangle:
    def __init__(self, width, height):
        self.__width = width
        self.__height = height

    def setWidth(self, width):
        self.__width = width
    def getWidth(self):
        return self.__width
    def setHeight(self, height):
        self.__height = height
    def getHeight(self):
        return self.__height

    def getArea(self):
        return self.__height * self.__width